json-endpoints/class.wpcom-json-api-bulk-update-comments-endpoint.php

```diff
@@ -0,0 +1,223 @@
+		'headers' => array(
+			'authorization' => 'Bearer YOUR_API_TOKEN'
+		),
+		'body' => array(
+			'comment_ids' => array( 881, 882 ),
+			'status'      => 'approved',
+		),
+	)
+) );
+
+new WPCOM_JSON_API_Bulk_Update_Comments_Endpoint( array(
+	'description'          => 'Permanently delete multiple comments. Note: this request will send non-trashed comments to the trash. Trashed comments will be permanently deleted.',
+	'group'                => 'comments',
+	'stat'                 => 'comments:1:bulk-delete',
+	'min_version'          => '1',
+	'max_version'          => '1',
+	'method'               => 'POST',
+	'path'                 => '/sites/%s/comments/delete',
+	'path_labels'          => array(
+		'$site' => '(int|string) Site ID or domain',
+	),
+	'request_format'       => array(
+		'comment_ids'  => '(array|string) An array, or comma-separated list, of Comment IDs to delete or trash. (optional)',
+		'empty_status' => '(string) Force to permanently delete all spam or trash comments. (optional). Allowed values: spam, trash',
+	),
+	'response_format'      => array(
+		'results' => '(array) An array of deleted or trashed Comment IDs.'
+	),
+	'example_request'      => 'https://public-api.wordpress.com/rest/v1.1/sites/82974409/comments/delete',
+	'example_request_data' => array(
+		'headers' => array(
+			'authorization' => 'Bearer YOUR_API_TOKEN'
+		),
+		'body' => array(
+			'comment_ids' => array( 881, 882 ),
+		),
+	)
+) );
+
+class WPCOM_JSON_API_Bulk_Update_Comments_Endpoint extends WPCOM_JSON_API_Endpoint {
+	// /sites/%s/comments/status
+	// /sites/%s/comments/delete
+	function callback( $path = '', $blog_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		$input = $this->input();
+
+		if ( is_array( $input['comment_ids'] ) ) {
+			$comment_ids = (array) $input['comment_ids'];
+		} else if ( ! empty( $input['comment_ids'] ) ) {
+			$comment_ids = explode( ',', $input['comment_ids'] );
+		} else {
+			$comment_ids = array();
+		}
+
+		$result = array(
+			'results' => array(),
+		);
+
+		wp_defer_comment_counting( true );
+
+		if ( $this->api->ends_with( $path, '/delete' ) ) {
+			if ( ! empty( $input['empty_status'] ) ) {
+				$empty_status = $input['empty_status'];
+				$result['results'] = $this->delete_all( $empty_status );
+			} else {
+				$result['results'] = $this->bulk_delete_comments( $comment_ids );
+			}
+		} else {
+			$status = $input['status'];
+			$result['results'] = $this->bulk_update_comments_status( $comment_ids, $status );
+		}
+
+		wp_defer_comment_counting( false );
+
+		return $result;
+	}
+
+	/**
+	 * Determine if the passed comment status is valid or not.
+	 *
+	 * @param string $status
+	 *
+	 * @return boolean
+	 */
+	function validate_status_param( $status ) {
+		return in_array( $status, array( 'approved', 'unapproved', 'pending', 'spam', 'trash' ) );
```

Comment:
By default `in_array` will also do type conversion before comparison if necessary, which could sometimes lead to unexpected results. For example, `in_array( true, array( 'approved', 'unapproved', 'pending', 'spam', 'trash' ) )` will evaluate to `true`. That's why I prefer passing `true` as third argument to force the strict comparison.